pkg/catalogsourceconfig/configuring.go

```diff
@@ -204,10 +205,21 @@ func newConfigMap(csc *v1alpha1.CatalogSourceConfig, data map[string]string) *co
 
 // newCatalogSource returns a CatalogSource object.
 func newCatalogSource(csc *v1alpha1.CatalogSourceConfig, configMapName string) *olm.CatalogSource {
-	return new(CatalogSourceBuilder).
-		WithMeta(v1alpha1.CatalogSourcePrefix+csc.Name, csc.Spec.TargetNamespace).
+	builder := new(CatalogSourceBuilder).
 		WithOwner(csc).
 		// TBD: where do we get display name and publisher from?
-		WithSpec("internal", configMapName, csc.Name, csc.Name).
-		CatalogSource()
+		WithSpec("internal", configMapName, csc.Name, csc.Name)
+
+	// Check if the operatorsource.DatastoreLabel is "true" which indicates that
+	// the CatalogSource is the datastore for an OperatorSource. This is a hint
+	// for us to set the OLM visibility label in the CatalogSource so that it is
+	// not visible in the OLM Packages UI.
+	labels := csc.ObjectMeta.GetLabels()
+	datastore, found := labels[operatorsource.DatastoreLabel]
```

Comment:
Where is the Datastore label being set? Is this going to be added to our example `OperatorSource` CR?

Reply:
No, this will be added in the `OperatorSource` `Configuring` reconciler.